core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -155,21 +155,47 @@ public ClientLogger asError() {
      */
     public void log(String format, Object... args) {
         if (canLogAtLevel(level)) {
-            performLogging(format, args);
+            performLogging(format, true, args);
         }
     }
 
+    /**
+     * This will log, if error log level is enabled , and throw the runtime exception.
+     * @param runtimeException to be thrown. It will do nothing if {@code null} is provided.
+     * @throws RuntimeException which is requested by this call.
+     */
+    public void logAndThrow(RuntimeException runtimeException) {
+        if (runtimeException == null) {
+            return;
+        }
+
+        // The default log level is DISABLED_LEVEL.  User's intention is to log and throw. Thus we are overriding
+        // level with ERROR_LEVEL if was is DISABLED_LEVEL.
+        int currentLevel = level;
+        if (level == DISABLED_LEVEL) {
+            level = ERROR_LEVEL;
+        }
+
+        // it will only log if log level is enabled in configuration
+        if (canLogAtLevel(level)) {
+            performLogging(runtimeException.getClass().getName(), false, runtimeException);
+        }
+        
+        // Change level back to what it was.
+        level = currentLevel;
```

Comment:
I'm not convinced we need this extra complexity. Could we just call `logger.error("", runtimeException)` without changing level or modifying any of the other code?

Reply:
I did had question myself on this, It is good suggestion to use logger.error.